texar/evals/bleu.py

```diff
@@ -196,15 +201,17 @@ def corpus_bleu(list_of_references: List[List[MaybeList[str]]],
     else:
         geo_mean = 0
 
-    ratio = float(hypothesis_length) / reference_length
-
-    if ratio > 1.0:
-        bp = 1.
+    if use_bp:
+        ratio = float(hypothesis_length) / reference_length
+        if ratio > 1.0:
+            bp = 1.
+        else:
+            try:
+                bp = math.exp(1 - 1. / ratio)
+            except ZeroDivisionError:
```

Comment:
This try-catch is really inelegant (and potentially a tiny bit slower). Let's just do a check on `abs(ratio) < 1e-8`.